nixos/modules/services/matrix/mautrix-discord.nix

```diff
@@ -0,0 +1,467 @@
+
+            User = "mautrix-discord";
+            Group = "mautrix-discord";
+
+            SystemCallFilter = [ "@system-service" ];
+
+            ProtectSystem = "strict";
+            ProtectHome = true;
+
+            ReadWritePaths = fullDataDir cfg;
+            StateDirectory = cfg.dataDir;
+            EnvironmentFile = cfg.environmentFile;
+          };
+
+          restartTriggers = [ (settingsFileUnsubstituted cfg) ];
+        };
+
+        mautrix-discord = {
+          description = "Mautrix-Discord bridge";
+          wantedBy = [ "multi-user.target" ];
+          wants = [ "network-online.target" ] ++ cfg.serviceDependencies;
+          after = [ "network-online.target" ] ++ cfg.serviceDependencies;
+
+          serviceConfig = {
+            Type = "simple";
+
+            User = "mautrix-discord";
+            Group = "mautrix-discord";
+            PrivateUsers = true;
+
+            LockPersonality = true;
+            MemoryDenyWriteExecute = true;
+            NoNewPrivileges = true;
+            PrivateDevices = true;
+            PrivateTmp = true;
+            ProtectClock = true;
+            ProtectControlGroups = true;
+            ProtectHome = true;
+            ProtectHostname = true;
+            ProtectKernelLogs = true;
+            ProtectKernelModules = true;
+            ProtectKernelTunables = true;
+            ProtectSystem = "strict";
+            Restart = "on-failure";
+            RestartSec = "30s";
+            RestrictRealtime = true;
+            RestrictSUIDSGID = true;
+            SystemCallArchitectures = "native";
+            SystemCallErrorNumber = "EPERM";
+            SystemCallFilter = [ "@system-service" ];
+            UMask = 27;
+
+            WorkingDirectory = fullDataDir cfg;
+            ReadWritePaths = fullDataDir cfg;
+            StateDirectory = cfg.dataDir;
+            EnvironmentFile = cfg.environmentFile;
+
+            ExecStart = lib.escapeShellArgs [
+              (lib.getExe cfg.package)
+              "--config=${settingsFile cfg}"
+            ];
+          };
+          restartTriggers = [ (settingsFileUnsubstituted cfg) ];
+        };
+      };
+      services.mautrix-discord = {
+        serviceUnit = "mautrix-discord.service";
+        registrationServiceUnit = "mautrix-discord-registration.service";
+        registrationFile = (fullDataDir cfg) + "/discord-registration.yaml";
+        settings =
+          let
+            inherit (lib.modules) mkDefault;
+          in
+          {
+            bridge = {
+              username_template = mkDefault "discord_{{.}}";
+            };
+
+            appservice = {
+              id = mkDefault "discord";
+              port = mkDefault 29334;
+              bot = {
+                username = mkDefault "discordbot";
+                displayname = mkDefault "Discord bridge bot";
+                avatar = mkDefault "mxc://maunium.net/nIdEykemnwdisvHbpxflpDlC";
+              };
+            };
+          };
```

Comment:
The `default` setting for `services.mautrix-discord.settings` sets all these fields to "". 

I think it'd be clearer if you just moved these defaults there instead of defining it down here in the config section.